src/DotLiquid/StandardFilters.cs

```diff
@@ -539,7 +539,7 @@ public static string Replace(Context context, string input, string @string, stri
             if (context.SyntaxCompatibilityLevel >= SyntaxCompatibility.DotLiquid21)
                 return input.Replace(@string, replacement);
 
-            return Regex.Replace(input, @string, replacement, RegexOptions.None, Template.RegexTimeOut);
+            return ExtendedFilters.RegexReplace(input, @string, replacement);

```

Comment:
nit: because input string and replacement are all of the same type (string), the arguments could be out of order (they aren't here). Hence it could reduce error risk by using named parameters, here and in the actual Regex Replace method call.